lib/common.js

```diff
@@ -35,6 +36,35 @@ const normalizeRequestOptions = function(options) {
   return options
 }
 
+/**
+ * Turns a URL into options. Taken directly from Node.js internals/url.js
+ * - Utility function that converts a URL object into an ordinary
+ *   options object as expected by the http.request and https.request
+ *   APIs.
+ * - https://github.com/nodejs/node/blob/7237eaa3353aacf284289c8b59b0a5e0fa5744bb/lib/internal/url.js#L1257
```

Comment:
👍 

@gr2m Since this is an internal Node API I think it makes sense to vendor in this function. Do you know if we need to acknowledge it somewhere besides what's done here, in order to comply with the license?